File: 3759-count-elements-with-at-least-k-greater-values/3759-count-elements-with-at-least-k-greater-values.py
class Solution:
    def countElements(self, nums: List[int], k: int) -> int:
        n = len(nums)
        if k == 0: return 0
        f = Counter(nums)
        li = sorted(list(f.keys()))
        res = 0
        for i in li:
            n -= f[i]
            if n >= k:
                res += f[i]
            else:
                break
        return res

File: 3759-count-elements-with-at-least-k-greater-values/3759-count-elements-with-at-least-k-greater-values.java
class Solution {
    public int countElements(int[] nums, int k) {
        int n = nums.length;
        if (k == 0) return n;
        Map<Integer, Integer> map = new HashMap<>();
        for (int i: nums) {
            map.put(i, map.getOrDefault(i,0) + 1);
        }
        List<Integer> ue = new ArrayList<>(map.keySet());
        Collections.sort(ue);

        int rm = n;
        int res = 0;

        for (int x: ue) {
            rm -= map.get(x);
            if (rm >= k) res += map.get(x);
            else break;
        }

        return res;
    }
}